Data_Structures_Questions.md

```diff
@@ -3,51 +3,56 @@ Answer the following questions for each of the data structures you implemented a
 ## Queue
 
 1. What is the runtime complexity of `enqueue`?
-
+   O(1)
 2. What is the runtime complexity of `dequeue`?
-
+   O(1)
 3. What is the runtime complexity of `len`?
+   O(n)
 
 ## Binary Search Tree
 
-1. What is the runtime complexity of `insert`? 
-
+1. What is the runtime complexity of `insert`?
+   O(log n)
 2. What is the runtime complexity of `contains`?
-
-3. What is the runtime complexity of `get_max`? 
+   O(log n)
+3. What is the runtime complexity of `get_max`?
+   O(log n)
 
 ## Heap
 
 1. What is the runtime complexity of `_bubble_up`?
-
+   O(logn)
 2. What is the runtime complexity of `_sift_down`?
-
+   O(logn)
 3. What is the runtime complexity of `insert`?
-
+   (logn) constant(append) + bubble up
 4. What is the runtime complexity of `delete`?
-
+   O(logn)
 5. What is the runtime complexity of `get_max`?
+   O(1)
 
 ## Doubly Linked List
 
 1. What is the runtime complexity of `ListNode.insert_after`?
-
+   O(1)
 2. What is the runtime complexity of `ListNode.insert_before`?
-
+   O(1)
 3. What is the runtime complexity of `ListNode.delete`?
-
+   O(1)
 4. What is the runtime complexity of `DoublyLinkedList.add_to_head`?
-
+   O(1)
 5. What is the runtime complexity of `DoublyLinkedList.remove_from_head`?
-
+   O(1)
 6. What is the runtime complexity of `DoublyLinkedList.add_to_tail`?
-
+   O(1)
 7. What is the runtime complexity of `DoublyLinkedList.remove_from_tail`?
-
+   0(1)
 8. What is the runtime complexity of `DoublyLinkedList.move_to_front`?
-
+   0(1)
 9. What is the runtime complexity of `DoublyLinkedList.move_to_end`?
-
+   O(1)
 10. What is the runtime complexity of `DoublyLinkedList.delete`?
+    O(1)
+    a. Compare the runtime of the doubly linked list's `delete` method with the worst-case runtime of the JS `Array.splice` method. Which method generally performs better?
 
-    a. Compare the runtime of the doubly linked list's `delete` method with the worst-case runtime of the JS `Array.splice` method. Which method generally performs better?
\ No newline at end of file
+    DLL runtime for delete is 0(1) whereas array.splice worst case is O(n) (because its possible that it has to copy over all the elements). Therefore the delete performs better in terms if runtime.
```

Comment:
Well done well done. It's funny though, even though DLL are so much better than arrays for adding, deleting, and whatnot, they are not utilized like arrays and sets. Probably because to find something in the middle requires so much work in general, compared to just having the array index and jumping straight to what you need to use.